cmd/dep/glide_importer.go

```diff
@@ -202,7 +202,8 @@ func (g *glideImporter) buildProjectConstraint(pkg glidePackage) (pc gps.Project
 	}
 	pc.Constraint, err = g.sm.InferConstraint(pkg.Reference, pc.Ident)
 	if err != nil {
-		return
+		g.logger.Printf("Unable to interpret revision specifier '%s' for package %s: %s", pkg.Name, pc.Ident, err.Error())
```

Comment:
Should these have trailing newlines `\n`?

I also wonder if it would be helpful to explicitly mention that `Any` is being used instead, or if that is the implied default or sufficiently documented elsewhere.

Reply:
Yup, the default for the importers is to use the any constraint whenever they can't determine something more specific. This ensures that the project is still listed in the manifest for direct dependencies.

This behavior is not documented because we don't currently doc how the importers work. I am knee deep in redoing the importer plumbing, and when I submit that PR, will add doc for the general heuristic used by all importers.